contracts/echidna/TestBond.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.9;
+
+import {Bond} from "../Bond.sol";
+import {TestERC20} from "../test/TestERC20.sol";
+
+contract TestBond {
+    event AssertionFailed(string);
+    Bond public bond;
+
+    TestERC20 public paymentToken;
+    TestERC20 public collateralToken;
+
+    uint256 public constant MAX_SUPPLY = 50000000 ether;
+    uint256 internal constant COLLATERAL_RATIO = 1.5 ether;
+    uint256 internal constant CONVERTIBLE_RATIO = .5 ether;
+    uint256 internal constant ONE = 1e18;
+    uint256 internal constant MAX_INT = 2**256 - 1;
+    uint256 internal maturityDate;
+
+    constructor() {
+        maturityDate = block.timestamp + 365 days;
+
+        paymentToken = new TestERC20("PT", "PT", MAX_INT, 6);
+        collateralToken = new TestERC20("CT", "CT", MAX_INT, 18);
+
+        bond = new Bond();
+        try
+            bond.initialize(
+                "bondName",
+                "bondSymbol",
+                address(this),
+                maturityDate,
+                address(paymentToken),
+                address(collateralToken),
+                COLLATERAL_RATIO,
+                CONVERTIBLE_RATIO,
+                MAX_SUPPLY
+            )
+        {} catch Error(string memory reason) {
+            emit AssertionFailed(reason);
+        }
+
+        /**
+            The BondFactory withdraws this collateral when creating a bond under
+            normal circumstances, but we are initializing the bond directly here.
+        */
+        collateralToken.transfer(
+            address(bond),
+            (MAX_SUPPLY * COLLATERAL_RATIO) / ONE
+        );
+        paymentToken.transfer(address(this), MAX_SUPPLY);
+    }
+
+    function approvePaymentTokenForBond() public {
+        paymentToken.approve(address(bond), MAX_INT);
+    }
+
+    function convertBonds(uint256 amount) public {
```

Comment:
Is this the below situation what that would be trying to avoid? 

The current convertibleRatio is 1:1 (collateral:bond). The collateral starts out worth 50 cents.  After a few days the collateral goes to 3 dollars. The bondholder sees an opportunity to convert all of their bonds into the collateral for a profit of $2.5 each. They create a convert transaction with all of their bonds @ 10 gwei/gas. 

Gas has not been 10 gwei for a week so their transaction has not been executed yet. 
One week later, gas drops below 10gwei and their transaction gets picked up. However, one collateral token is now worth 10 cents. They end up converting their bonds for a loss of 90 cents each!